security/subsystem/src/main/java/org/jboss/as/security/service/SimpleSecurityManager.java

```diff
@@ -323,37 +323,37 @@ public void push(final String securityDomain) {
         RunAs currentRunAs = current.getIncomingRunAs();
         boolean trusted = currentRunAs != null && currentRunAs instanceof RunAsIdentity;
 
-        if (trusted == false) {
+        if (trusted && SecurityActions.remotingContextIsSet()) {
```

Comment:
I am just quickly scanning the diff so may have missed some lines but this seems to have lost the "trusted==false"?

Reply:
Thanks for the quick reply. I saw the failing test and it is resolved now.